POSIX/posix-abstract-declarator.c

```diff
@@ -0,0 +1,17 @@
+/* To trigger the problem, we must not have an argument name. */
+static int func(int [static restrict 1]);
```

Comment:
How about making this `static int func(int ARGNAME[static restrict 1]);` and testing not just whether it fails to compile with ARGNAME defined to "" but also checking that it compiled with ARGNAME *not* defined to ""?

Otherwise this test could fail for the wrong reason (e.g. old compilers which don't understand "restrict").

Reply:
hmm, ok.  I put this test after the `restrict` test in `posix-cflags.sh`, but your method is safer.